File: app/src/main/java/com/specknet/pdiotapp/history/Activity.java
package com.specknet.pdiotapp.history;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.annotation.NonNull;

import java.time.LocalDateTime;
import java.time.Period;

/**
 * A basic class representing an entity that is a row in a three-column database table.
 * Every row is one activity that the user has carried out.
 * <p>
 * @ Entity - You must annotate the class as an entity and supply a table name if not class name.
 * @ PrimaryKey - You must identify the primary key.
 * @ ColumnInfo - You must supply the column name if it is different from the variable name.
 * <p>
 * See the documentation for the full rich set of annotations.
 * <a href="https://developer.android.com/topic/libraries/architecture/room.html">...</a>
 */


@Entity(tableName = "activity_table")
public class Activity {
    @NonNull
    public Long getStart() {
        return start;
    }

    @NonNull
    public String getModel() {return model; }

    @NonNull
    public String getActivity_type() {
        return activity_type;
    }

    public long getId() {
        return id;
    }

    @NonNull
    @ColumnInfo(name = "Start")
//    private LocalDateTime start;
    private Long start;



    @NonNull
    @ColumnInfo(name = "Type")
    private String activity_type;  // TODO: enum?

    @NonNull
    @ColumnInfo(name = "Model")
    private String model;

    // TODO: activity type vs name.

//    @NonNull
//    @ColumnInfo(name = "Duration")
////    private Period duration;
//    private String duration;

    public Activity(
//            @NonNull LocalDateTime start,
//            @NonNull LocalDateTime end,
            @NonNull Long start,
            @NonNull String activity_type,
            @NonNull String model
//            @NonNull Period duration
//            @NonNull String duration
    ) {
        this.start = start;
        this.activity_type = activity_type;
        this.model = model;
//        this.duration = duration;
    }

    // TODO: check whether this actually  auto-generates
    @PrimaryKey(autoGenerate = true)
    long id = 0;


//    @NonNull
//    public String getWord() {
//        return this.mWord;
//    }

    @NonNull
    public String getActivity() {
        return (this.start.toString() + this.activity_type + this.model);
    }
}

File: app/src/main/java/com/specknet/pdiotapp/history/Activity.kt
//package com.specknet.pdiotapp.history
//
//import androidx.room.ColumnInfo
//import androidx.room.Entity
//import androidx.room.PrimaryKey
//import java.util.Date
//
//@Entity
//data class Activity(
//    // TODO: use @ColumnInfo() so that you can change column name without changing class var names
//
//    @PrimaryKey(autoGenerate = true) val uid: Int,
//    // TODO: Delete later
//    @ColumnInfo(name = "first_name") val firstName: String?,
//    @ColumnInfo(name = "last_name") val lastName: String?,
//
//    @ColumnInfo(name = "activity_date") val activityDate: String,  // TODO: change to DATE
//    @ColumnInfo(name = "activity_time") val activityTime: String,  // TODO: change to TIME
//    @ColumnInfo(name = "activity_name") val activityName: String,  // TODO: change to Enum - String
//    // @ColumnInfo(name = "activity_type")
//    @ColumnInfo(name = "activity_duration") val activityDuration: String  // TODO: change to DURATION
//)